### PR TITLE
[CSBindings] Don't generate bindings for defaults

### PR DESCRIPTION
**Pull request by xedin**
Let's keep defaults separate from direct and transitive bindings,
that would make it easier to handle them in incremental model.

Instead of generating bindings for defaults and adding to the main
set, let's allow producer to choose what to do with them once type
variable has been picked for attempting.

### CODE REVIEW

File: include/swift/Sema/ConstraintSystem.h [hunk 1]
```diff
@@ -4838,19 +4839,38 @@ class ConstraintSystem {
...
+        return 1;
+
+      // We might want to consider adding this as a field to `PotentialBindings`
+      // and collect this information as new bindings become available but,
+      // since we are moving towards incremental model, which implies that
+      // `PotentialBindings` would stay alive for a long time, it's not
+      // immediately clear whether storage overhead of keeping this set just
+      // for ranking is worth it.
+      llvm::SmallPtrSet<CanType, 4> discoveredTypes;
+      for (const auto &binding : Bindings)
```
Comment (xedin) [inline include/swift/Sema/ConstraintSystem.h:4853]: After looking at this again, I think I can make it work without having to use a separate set.

Comment (xedin) [reply include/swift/Sema/ConstraintSystem.h]: Ok, done!

**Comment by xedin** [comment]
@swift-ci please test source compatibility

**Comment by xedin** [comment]
@swift-ci please test source compatibility

**Comment by xedin** [comment]
@shahmishal There is a problem with one of the projects in compat suite:

```
fatal: unable to access 'https://github.com/andreamazz/AMScrollingNavbar.git/': Could not resolve host: github.com
```

This might be an account issue because I can access it with mine.

**Comment by xedin** [comment]
@swift-ci please test source compatibility

**Comment by xedin** [comment]
@swift-ci please smoke test

**Review by hborla — APPROVED** [review]

**Pull request merged**